### ISSUE TITLE
eclipseMavenCentral does not convert "org.eclipse.jdt.core.compiler.batch" correctly

### ISSUE THREAD

**Issue by jmini** [diffplug/goomph#157]
Probably due to legacy reasons it was decided to publish bundle `org.eclipse.jdt.core.compiler.batch` as:

```xml
<dependency>
    <groupId>org.eclipse.jdt</groupId>
    <artifactId>ecj</artifactId>
    <version>${jdtVersion}</version>
</dependency>
```

---
This special mapping is defined in the `eclipse.platform.releng` project in the CBI aggregator configuration file:

```xml
<mavenMappings namePattern="(org\.eclipse\.jdt)\.core\.compiler\.batch" groupId="$1" artifactId="ecj"/>
```
Source: https://git.eclipse.org/c/platform/eclipse.platform.releng.git/tree/publish-to-maven-central/SDK4Mvn.aggr#n36

---

Currently `eclipseMavenCentral` is now aware of this special mapping, it is failing with:

```
> Task :my_project:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':my_project:compileJava'.
> Could not resolve all files for configuration ':my_project:compileClasspath'.
   > Could not find org.eclipse.jdt:org.eclipse.jdt.core.compiler.batch:3.20.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/org/eclipse/jdt/org.eclipse.jdt.core.compiler.batch/3.20.0/org.eclipse.jdt.core.compiler.batch-3.20.0.pom
     Required by:
         project :my_project
```

**Comment by jmini** [comment]
I think this special mapping should be hard-coded here:

https://github.com/diffplug/goomph/blob/f1d8086d792234e4c56aaf892b0deaf29315db0e/src/main/java/com/diffplug/gradle/eclipse/MavenCentralMapping.java#L53-L63

If you agree, I will propose a PR.

**Comment by nedtwigg** [comment]
I agree, PR welcome!

**Comment by nedtwigg** [comment]
Fixed in `3.32.2`
